### PR TITLE
Mobx examples

### PR DESCRIPTION
**Pull request by pavanpodila**
A collection of examples from simple to complex to show case the power of MobX

### CODE REVIEW

File: mobx_examples/lib/todos/todo_widgets.dart [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_mobx/flutter_mobx.dart';
+
+import 'todo_list.dart';
+
+final TodoList list = TodoList();
```
Comment (katis) [inline mobx_examples/lib/todos/todo_widgets.dart]: could use the name `todos` instead of `list`?

File: mobx_examples/lib/todos/todo_list.dart [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import 'package:mobx/mobx.dart';
+import 'package:mobx_examples/todos/todo.dart';
+
+part 'todo_list.g.dart';
+
+enum VisibilityFilter { all, pending, completed }
+
+class TodoList = TodoListBase with _$TodoList;
+
+abstract class TodoListBase implements Store {
```
Comment (katis) [inline mobx_examples/lib/todos/todo_list.dart]: Maybe use the name `_TodoList`?

File: mobx_examples/lib/todos/todo.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import 'package:mobx/mobx.dart';
+
+part 'todo.g.dart';
+
+class Todo = TodoBase with _$Todo;
+
+abstract class TodoBase implements Store {
```
Comment (katis) [inline mobx_examples/lib/todos/todo.dart]: Could just be `_Todo`.

File: mobx_codegen/lib/src/template/store.dart [hunk 4]
```diff
@@ -27,6 +27,8 @@ class StoreTemplate {
 
   @override
   String toString() => """
+  // ignore_for_file: non_constant_identifier_names, unnecessary_lambdas, prefer_expression_function_bodies
```
Comment (pavanpodila) [inline mobx_codegen/lib/src/template/store.dart]: Adding these ignores as the generated files should be lint free.

**Pull request merged**